src/ccid_usb.c

```diff
@@ -644,6 +644,39 @@ status_t OpenUSBByName(unsigned int reader_index, /*@null@*/ char *device)
 				}
 #endif
 
+#ifdef USE_COMPOSITE_AS_MULTISLOT
+				if (VENDOR_KAPELSE == GET_VENDOR(readerID))
+				{
+					switch (readerID)
+				 	{
+						case KAPELSE_KAPECV :
+						    /* KAP-eCV : only first interface is a ccid one */
+						    max_interface_number=0;
+						    DEBUG_INFO1("Kapelse reader forced as monoslot!");
+						    break;
+						    
+						case KAPELSE_KAPLIN2 :
+						    /* KAP&LINK2 : only 3 first interfaces are ccid ones */
+						    if(config_desc->bNumInterfaces > 3)
```

Comment:
Why this test?
In my list the KAP-LINK2 has 3 CCID interfaces https://ccid.apdu.fr/ccid/shouldwork.html#0x29470x0105

Do you have another model with a different configuration but using the same `idProduct`?

Reply:
This test is needed as KAP-LINK2 is a USB composite device presenting either 3 CCID interfaces or 3 CCID interfaces (the 3 first ones) and a CDC-ACM interface according user configuration

Reply:
I see.
It should work fine if you something like use:
```
max_interface_number = 3; /* up to 4 interfaces */
num_CCID_interfaces = 3;  /* 3 CCID interfaces */
```

Reply:
@bmoraine Is it possible for the KAP-LINK2 to NOT have 3 CCID interfaces?

Why can't we use ``max_interface_number = 2;`` in all cases without checking the value of ``config_desc->bNumInterfaces``?

Reply:
@LudovicRousseau Effectively, KAP-LINK2 has always 3 CCID interfaces (the 3 first ones) and an optional 4th CDC-ACM interface. So, you are right, we can use max_interface_number = 2; in all cases without checking the value of config_desc->bNumInterfaces